include/git2/submodule.h

```diff
@@ -223,6 +223,15 @@ GIT_EXTERN(int) git_submodule_lookup(
 	git_repository *repo,
 	const char *name);
 
+/**
+ * Create an in-memory copy of a submodule. The copy must be explicitly
+ * free'd or it will leak.
+ *
+ * @param out Pointer to store the copy of the submodule. Cannot be NULL.
```

Comment:
Our docs generally tell you when a param is optional, instead of when a param is required.

```suggestion
 * @param out Pointer to store the copy of the submodule.
```

Reply:
@ethomson 
Fixed